### DISCUSSION TITLE
Question: How can I get a function source in my `DebugFrame` implementation?

### DISCUSSION THREAD

**Discussion by avserban** [mozilla/rhino#1137] [Q&A]
Hi,

I think it is appropriate for me to start this thread like so: Merry Christmas!

I just wanted to ask a question as I am stuck with something while implementing my very own `DebugFrame` using this framework.

My question: I was wondering if I could get the source string of a function inside my class that implements `DebugFrame`?

What I have tried so far: I have tried to get the source string of the executing function inside of the `onEnter(Context cx, Scriptable activation, Scriptable thisObj, Object[] args)` function by using the various parameters that are passed down (such as `activation`), but not luck...

Things worth mentioning:
* Inside my own implementation of `DebugFrame` I am keeping a reference to the `DebuggableScript` related to the function... maybe I could use this to get the source string of the function?

I can imagine what I am asking might be super easy to achieve, however I am quite new to the framework itself, therefore any help is greatly appreciated

Thank you in advance for your help on the matter.

**Comment by p-bakker** [comment]
Have you looked at the debugger implementation that is contained within the repo? See https://github.com/mozilla/rhino/tree/master/toolsrc/org/mozilla/javascript/tools/debugger and more specifically the Dim class

**Reply by p-bakker** [reply]
Another example you might look at is mentioned in https://github.com/mozilla/rhino/issues/973#issuecomment-917405381
